metaflow/plugins/argo/argo_workflows.py

```diff
@@ -176,32 +194,63 @@ def trigger(cls, name, parameters=None):
         except Exception as e:
             raise ArgoWorkflowsException(str(e))
 
-    def _get_cron(self):
+    def _get_schedule(self):
         schedule = self.flow._flow_decorators.get("schedule")
         if schedule:
             # Remove the field "Year" if it exists
             schedule = schedule[0]
             return " ".join(schedule.schedule.split()[:5]), schedule.timezone
-        return None
+        return None, None
 
     def schedule(self):
         try:
-            if self._cron is None:
-                cron, timezone = None, None
-            else:
-                cron, timezone = self._cron
             ArgoClient(namespace=KUBERNETES_NAMESPACE).schedule_workflow_template(
-                self.name, cron, timezone
+                self.name, self._schedule, self._timezone
+            )
+            # Register sensor. Unfortunately, Argo Events Sensor names don't allow for
+            # dots (sensors run into an error) which rules out self.name :(
+            # Metaflow will overwrite any existing sensor.
+            ArgoClient(namespace=KUBERNETES_NAMESPACE).register_sensor(
+                self.name.replace(".", "-"),
+                self._sensor.to_json() if self._sensor else {},
             )
         except Exception as e:
             raise ArgoWorkflowsSchedulingException(str(e))
 
     def trigger_explanation(self):
-        if self._cron:
+        # Trigger explanation for cron workflows
+        if self.flow._flow_decorators.get("schedule"):
             return (
                 "This workflow triggers automatically via the CronWorkflow *%s*."
                 % self.name
             )
+
+        # Trigger explanation for @trigger
+        elif self.flow._flow_decorators.get("trigger"):
+            return (
+                "This workflow triggers automatically when the upstream %s "
+                "is/are published."
+                % self.list_to_prose(
+                    [event["name"] for event in self.triggers], "event"
+                )
+            )
+
+        # Trigger explanation for @trigger_on_finish
+        elif self.flow._flow_decorators.get("trigger_on_finish"):
+            msg = (
+                "This workflow triggers automatically when the upstream %s succeed(s)"
```

Comment:
Due to triggers relying on the webhook, this is true if the upstream flow has also been deployed with events enabled, and is an argo-workflow (so it auto-emits).

This can be bit of a gotcha for users that have existing deployed flows that they try to use downstream with `trigger_on_finish`, but they never emit the necessary event.

I would either add a mention here that the upstream also needs to have been deployed with an event-capable version of metaflow, or make note of the requirement in the release notes

Reply:
+1